### ISSUE TITLE
tf.keras.backend.temporal_padding crashes(bad_alloc) when padding is large

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#42293]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
`tf.keras.backend.temporal_padding` crashes(abort, bad_alloc) when `padding` is large

**Describe the expected behavior**
Expect no crashes

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
~~~python

import tensorflow as tf
tf.keras.backend.temporal_padding(x=tf.ones((2,1,1)), padding=(6400000000000000000, 4314310000000000000))
~~~
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

In TF2.1
~~~python
  what():  std::bad_alloc
Aborted (core dumped)
~~~
In TF nightly
~~~
2020-08-13 01:04:13.669045: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2020-08-13 01:04:13.683160: F tensorflow/core/framework/tensor_shape.cc:345] Check failed: size >= 0 (-7732434073709551615 vs. 0)
Aborted (core dumped)

~~~

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.1, TF v2.3 and TF-nightly. 

Session crashes on running the `tf.keras.backend.temporal_padding` line. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/f7850e90c6c3500a02b4a555770e4bcd/42293.ipynb).

**Comment by qlzh727** [comment]
This is expected, since TF is trying to allocate the memory for a huge tensor with shape (2, 6400000000000000000 + 1 + 4314310000000000000, 1), and it will certainly OOM. 

I am closing this issue since I don't think it is working as intended. Feel free to reopen if you feel otherwise.
